DynamicSmoothedGUI.py

```diff
@@ -169,6 +171,56 @@ def _color_normally(r, angle_rad):
                    _color_normally(r, angle_rad)
                lidar_data_copy.pop(0)
 
+    def drawC1C0(self):
+        """Draws C1C0's current location on the simulation"""
+
+        def _scale_coords(coords):
+            """scales coords (a tuple (x, y)) from real life cm to pixels"""
+            scaled_x = coords[0] / tile_scale_fac
+            scaled_y = coords[1] / tile_scale_fac
+            scaled_coords = (scaled_x, scaled_y)
+            return scaled_coords
+
+        # coordinates of robot center right now (in cm)
+        center_x = self.curr_tile.x
+        center_y = self.curr_tile.y
+
+        # converting heading to radians, and adjusting so that facing right = 0 deg
+        heading_adj_rad = math.radians(self.heading + 90)
+
+        if self.prev_draw_c1c0_ids is not None:
+            # delete old drawings from previous iteration
+            self.canvas.delete(self.prev_draw_c1c0_ids[0])
+            self.canvas.delete(self.prev_draw_c1c0_ids[1])
+        # coordinates of bounding square around blue circle
+        top_left_coords = (center_x - robot_radius, center_y + robot_radius)
+        bot_right_coords = (center_x + robot_radius, center_y - robot_radius)
+        # convert coordinates from cm to pixels
+        top_left_coords_scaled = _scale_coords(top_left_coords)
+        bot_right_coords_scaled = _scale_coords(bot_right_coords)
+
+        self.prev_draw_c1c0_ids = [None, None]
```

Comment:
First I would consider making this a tuple? Since logically it is a tuple not a list you need to iterate through or change.

Second I would set this attribute to be this in the initializer, and then you can remove this LOC. 
Then each time you would just over-write the values in the array instead of creating a new array at each iteration

Reply:
i'm not sure i understand what you mean by that i don't need to iterate through or change the prev_draw_c1c0_ids. i do change the values in each iteration?

Reply:
Nice change, this way you aren't allocating space for a new array at each iteration. (I am being pedantic about little things sorry)

Also what I meant was that it a tuple is more logical here because you don't have to iterate through the prev_draw_c1c0_ids data structure, you always just have two entries you update. So while you do update your data structure at each new frame,  you do not actually iterate through it. A tuple represents a fixed sized array, but you can do everything with a list that you can do with a tuple.

But I am being pedantic again, a two element list is A OK